File: plugins/cordova_capacitor/cordova/src/android/AranRASP.java
/**
 * ARAN RASP ENGINE - Cordova Android Plugin (Abstract Architecture)
 * 
 * This is a professional, abstract RASP layer that acts as a "dumb" passthrough
 * to the pre-compiled native cores (AAR).
 * 
 * Security Requirements:
 * - NO SENSITIVE STRINGS in framework code
 * - USE OBFUSCATED SELECTORS (int values)
 * - "Dumb" passthrough to pre-compiled native cores
 * 
 * Architecture:
 * - Cordova (JS) -> Native Bridge -> Native Core (AAR)
 * - No logic in plugin - all security logic in native core
 */

package com.aran.rasp;

import org.apache.cordova.CordovaPlugin;
import org.apache.cordova.CallbackContext;
import org.apache.cordova.CordovaArgs;
import org.json.JSONArray;
import org.json.JSONException;
import com.aran.core.ARANRasp; // Fetched from your .aar

public class AranRASP extends CordovaPlugin {
    
    @Override
    public boolean execute(String action, JSONArray args, CallbackContext callbackContext) throws JSONException {
        if (action.equals("execute")) {
            int selector = args.getInt(0);
            // "Dumb" passthrough - the actual logic is in the AAR
            int result = ARANRasp.runAudit(selector);
            callbackContext.success(result);
            return true;
        } else if (action.equals("getStatus")) {
            int statusType = args.getInt(0);
            // "Dumb" passthrough - the actual logic is in the AAR
            int result = ARANRasp.getStatus(statusType);
            callbackContext.success(result);
            return true;
        } else if (action.equals("initialize")) {
            // "Dumb" passthrough - the actual logic is in the AAR
            ARANRasp.initialize();
            callbackContext.success();
            return true;
        } else if (action.equals("shutdown")) {
            // "Dumb" passthrough - the actual logic is in the AAR
            ARANRasp.shutdown();
            callbackContext.success();
            return true;
        }
        return false;
    }
}

File: plugins/cordova-plugin-aran-rasp/src/ios/AranRASP.m
/**
 * ARAN RASP ENGINE - Cordova iOS Plugin (Abstract Architecture)
 * 
 * This is a professional, abstract RASP layer that acts as a "dumb" passthrough
 * to the pre-compiled native cores (Pod).
 * 
 * Security Requirements:
 * - NO SENSITIVE STRINGS in framework code
 * - USE OBFUSCATED SELECTORS (int values)
 * - "Dumb" passthrough to pre-compiled native cores
 * 
 * Architecture:
 * - Cordova (JS) -> Native Bridge -> Native Core (Pod)
 * - No logic in plugin - all security logic in native core
 */

#import <Cordova/CDV.h>
#import <AranRuntime/AranRuntime.h>

@interface AranRASP : CDVPlugin

@end

@implementation AranRASP

- (void)execute:(CDVInvokedUrlCommand*)command {
    int selector = [[command.arguments objectAtIndex:0] intValue];
    
    // "Dumb" passthrough - the actual logic is in the Pod
    // Call the Mach-O binary directly
    NSInteger result = [AranRuntime performInternalAudit:(uint32_t)selector];
    
    CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsInt:(int)result];
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

- (void)initialize:(CDVInvokedUrlCommand*)command {
    // "Dumb" passthrough - the actual logic is in the Pod
    [AranRuntime initializeEngine];
    
    CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

- (void)shutdown:(CDVInvokedUrlCommand*)command {
    // "Dumb" passthrough - the actual logic is in the Pod
    [AranRuntime shutdownEngine];
    
    CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

- (void)getStatus:(CDVInvokedUrlCommand*)command {
    int statusType = [[command.arguments objectAtIndex:0] intValue];
    
    // "Dumb" passthrough - the actual logic is in the Pod
    NSInteger result = [AranRuntime getStatus:(uint32_t)statusType];
    
    CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsInt:(int)result];
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

@end
